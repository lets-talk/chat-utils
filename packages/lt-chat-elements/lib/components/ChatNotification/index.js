import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Flex, Box } from 'grid-styled';

import MdClose from 'react-icons/lib/md/close';

import AvatarGroup from '../AvatarGroup';
import ChatCard from '../ChatCard';
import { PersonType } from '../../utils/types';
import { textColor, themeColor } from '../../utils/style';


const ChatNotification = (props) => {
  const {
    showAvatars, persons, title, body, dismissText, avatarPosition, width, height,
  } = props;

  const StyledDismissButton = styled.button`
    display: flex;
    cursor: pointer;
    float: right;
    opacity: 0;
    padding: 8px 15px;
    border-radius: 100px;
    color: ${(innerProps) => textColor(innerProps.theme, 'dark', 'primary')};
    background-color: ${(innerProps) => themeColor(innerProps.theme, 'foreground', 'base')};
    font-size: ${(innerProps) => innerProps.theme.typography.classes.caption.fontSize};
    line-height: ${(innerProps) => innerProps.theme.typography.classes.caption.lineHeight};
    font-weight: ${(innerProps) => innerProps.theme.typography.weights.fontWeightMedium};
  
    span {
      margin-left: 10px;
    }
  `;

  const StyledNotificationContainer = styled.div`
    max-width: ${props.width}px;
    max-height: ${props.height}px;

    border-radius: ${() =>
    (props.type && props.type === 'rounded' && '5px 5px 0 0')
    || (props.type && props.type === 'circle' && '50%')
    || '0'};

    &:hover ${StyledDismissButton} {
      opacity: 1;
      visibility: visible;
    }

    &:focus {
      outline: none;
    }
  `;

  const StyledNotificationBody = styled.div`
    height: 100%;
    position: relative;
    clear: both;
`;

  const StyledChatSnippet = styled.div`
    height: 100%;
    position: relative;
  `;

  const avatarOutside = avatarPosition === 'out';

  return (
    <StyledNotificationContainer>
      <Flex>
        <Box p={1} width={1}>
          <StyledDismissButton>
            <div>
              {dismissText}
            </div>
            <span>
              <MdClose size={15} />
            </span>
          </StyledDismissButton>
        </Box>
      </Flex>
      <Flex>
        <Box p={1} m={2}>
          {showAvatars && avatarOutside &&
          <AvatarGroup avatars={persons} />
          }
        </Box>
        <Box pt={1} width={1}>
          <StyledNotificationBody>
            <StyledChatSnippet>
              <ChatCard
                width={width}
                height={height}
                title={title}
                body={body}
                showRenderProp={showAvatars && !avatarOutside}
                render={() => <AvatarGroup avatars={persons} groupType="line" />}
              />
            </StyledChatSnippet>
          </StyledNotificationBody>
        </Box>
      </Flex>
    </StyledNotificationContainer>
  );
};


ChatNotification.propTypes = {
  /**
   * types: default, rounded, circle
   */
  type: PropTypes.string,
  /**
   * title
   */
  title: PropTypes.string,
  /**
   * dismissText
   */
  dismissText: PropTypes.string,
  /**
   * body
   */
  body: PropTypes.string,
  /**
   * showAvatars: Wheter or not display the chat icon
   */
  showAvatars: PropTypes.bool,
  /**
   * avatarPosition: Wheter or not display the chat icon
   */
  avatarPosition: PropTypes.oneOf(['in', 'out']),
  /**
   * showAvatars: Wheter or not display the chat icon
   */
  persons: PropTypes.arrayOf(PersonType),
  /**
   * width: Set the width of the notification
   */
  width: PropTypes.number,
  /**
   * height: Set the height of the notification
   */
  height: PropTypes.number,
};

ChatNotification.defaultProps = {
  type: 'rounded',
  persons: [],
  showAvatars: true,
  avatarPosition: 'out',
  dismissText: 'Cerrar',
};

export default ChatNotification;
