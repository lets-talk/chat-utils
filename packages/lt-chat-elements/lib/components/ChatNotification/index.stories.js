import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, number, select, object } from '@storybook/addon-knobs';

import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';
// Component to show on storybook
import ChatNotification from './';

const stories = storiesOf('ChatNotification', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('ChatNotification'));

const avatars = [
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
];
// style={{
//   position: 'absolute', top: '50%', left: '50%', width: '100px', height: '100px', marginTop: '50px', marginLeft: '50px',
// }}
stories.addWithInfo(
  'ChatNotification',
  'This is the basic usage with the ChatNotification with providing a label to show the text.',
  () => (
    <div>
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <ChatNotification
          persons={object('Persons', avatars)}
          title={text('Text', "Let's Talk")}
          body={text('Body', 'Hi! Have any questions about pricing?')}
          dismissText={text('dismissText', 'Close')}
          width={number('Width', 350)}
          height={number('Height', 350)}
          margin={text('Margin', '0px 10px 0px 0px')}
          showAvatars={boolean('showAvatars', true)}
          avatarPosition={select('AvatarPosition', { in: 'in', out: 'out' }, 'out')}
          onClick={action('onClick')}
          onDismiss={action('onDismiss')}
        />
      </WrapWithTheme>
    </div>
  )
);
