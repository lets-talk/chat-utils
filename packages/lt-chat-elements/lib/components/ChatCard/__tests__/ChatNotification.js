import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChatNotification from '../index';

describe('ChatNotification component', () => {
  it('should render with no props', () => {
    const component = shallow(<ChatNotification />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      type: 'circle',
      text: '',
      showIcon: true,
      height: 50,
      width: 50,
      margin: '0px 10px 0px 0px',
      chat_icon_pic: '',
      display: 'minimized',
    };

    const component = shallow(<ChatNotification {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
