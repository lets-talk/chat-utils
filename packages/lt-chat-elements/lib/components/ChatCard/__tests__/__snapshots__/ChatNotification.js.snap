// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatNotification component should render with custom props 1`] = `
<styled.div>
  <Flex
    alignItems="center"
    flex="1 1 auto"
    pl={2}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <styled.div />
  </Flex>
  <Flex
    flex="1 1 auto"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      ml="auto"
      p={2}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <styled.div />
    </Box>
  </Flex>
</styled.div>
`;

exports[`ChatNotification component should render with no props 1`] = `
<styled.div>
  <Flex
    alignItems="center"
    flex="1 1 auto"
    pl={2}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <styled.div />
  </Flex>
  <Flex
    flex="1 1 auto"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      ml="auto"
      p={2}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <styled.div />
    </Box>
  </Flex>
</styled.div>
`;
